lib/date_time/date_time_core.c

```diff
@@ -53,27 +56,35 @@ static void date_time_core_notify_event(enum date_time_evt_type time_source)
 	}
 }
 
-static void date_time_core_schedule_update(bool check_pending)
+static void date_time_core_schedule_work(bool retry)
 {
-	/* (Re-)schedule time update work
-	 * if periodic updates are requested in the configuration.
-	 */
-	if (CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS > 0) {
-		/* Don't reschedule time update work in some cases,
-		 * e.g. if time is not obtained and the work is already pending.
-		 */
-		if (check_pending && k_work_delayable_is_pending(&date_time_update_work)) {
-			return;
-		}
+	const int delay = retry ? CONFIG_DATE_TIME_RETRY_INTERVAL_SECONDS :
+				  CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS;
+
+	LOG_DBG("New periodic date time update in: %d seconds%s", delay, retry ? " (retry)" : "");
 
-		LOG_DBG("New periodic date time update in: %d seconds",
-			CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS);
+	k_work_schedule_for_queue(&date_time_work_q, &date_time_update_work, K_SECONDS(delay));
```

Comment:
If the work has already been scheduled (and isn't re-scheduled), the above debug log contains incorrect time. The return value should be 1 if the work was scheduled by the call and 0 if it was already scheduled or submitted.

Furthermore, I think it would make the implementation a bit clearer if this function was removed and the scheduling of the work would be in `date_time_core_schedule_update()` and `date_time_core_schedule_retry()` functions. To figure out what the current `date_time_core_schedule_work()` call does, you have to check the meaning of the boolean parameter.

Reply:
> If the work has already been scheduled (and isn't re-scheduled), the above debug log contains incorrect time. The return value should be 1 if the work was scheduled by the call and 0 if it was already scheduled or submitted.

I had to switch away from this approach anyways, it turns out, but you are quite right. The new approach I think will not have this issue

> Furthermore, I think it would make the implementation a bit clearer if this function was removed and the scheduling of the work would be in date_time_core_schedule_update() and date_time_core_schedule_retry() functions. To figure out what the current date_time_core_schedule_work() call does, you have to check the meaning of the boolean parameter.

I will keep `date_time_core_schedule_work`, but remove the `bool retry` parameter and replace it with the interval directly.

Reply:
Alright, that is pushed. Hopefully, I think, you will find this preferable